### PR TITLE
changed: move CFileItem::FillInDefaultIcon to ArtUtils

### PR DESCRIPTION
**Pull request by notspiff**
## Description
Move another function from CFileItem to ArtUtils

## Motivation and context
Less code in CFileItem

## How has this been tested?
It builds + added tests passes.

## What is the effect on users?
None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [x] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by notspiff** [comment]
jenkins build this please

**Comment by fuzzard** [comment]
Jenkins build this please

File: xbmc/utils/ArtUtils.cpp [hunk 1]
```diff
@@ -28,6 +31,126 @@ using namespace XFILE;
...
+void FillInDefaultIcon(CFileItem& item)
+{
+  if (URIUtils::IsPVRGuideItem(item.GetPath()))
+  {
+    // epg items never have a default icon. no need to execute this expensive method.
+    // when filling epg grid window, easily tens of thousands of epg items are processed.
+    return;
+  }
+
+  //CLog::Log(LOGINFO, "FillInDefaultIcon({})", pItem->GetLabel());
```
Comment (ksooo) [inline xbmc/utils/ArtUtils.cpp:43]: Remove this line?

File: xbmc/utils/ArtUtils.cpp [hunk 2]
```diff
@@ -28,6 +31,126 @@ using namespace XFILE;
...
+    if (!item.m_bIsFolder)
+    {
+      /* To reduce the average runtime of this code, this list should
+       * be ordered with most frequently seen types first.  Also bear
+       * in mind the complexity of the code behind the check in the
+       * case of IsWhatever() returns false.
+       */
+      if (item.IsPVRChannel())
+      {
+        if (item.GetPVRChannelInfoTag()->IsRadio())
```
Comment (ksooo) [inline xbmc/utils/ArtUtils.cpp:67]: Seems this is the only reason for including "PVRChannel.h" (compile time dep to PVR component). Can you ple ase put a `@todo get rid of PVR compile time dependency` and I will pick this up later?

Comment (ksooo) [reply xbmc/utils/ArtUtils.cpp:67]: Or if you like, you could introduce `bool CFileItem::IsPVRRadioChannel() const`?

File: xbmc/utils/ArtUtils.cpp (hunk 1 above)
Comment (notspiff) [reply xbmc/utils/ArtUtils.cpp]: done.

File: xbmc/utils/ArtUtils.cpp (hunk 2 above)
Comment (notspiff) [reply xbmc/utils/ArtUtils.cpp:67]: added the todo.

**Review by ksooo — APPROVED** [review]
Shove it in.

**Pull request merged**